### ISSUE TITLE
Some tests in System.Net.HttpListener HttpRequestStreamTests.cs will break UWP test runner with System.IO.IOException

### ISSUE THREAD

**Issue by caesar1995** [dotnet/corefx#21953]
Those are sample test cases will break the test runner: `Read_NullBuffer_ThrowsArgumentNullException()`, `Read_InvalidOffset_ThrowsArgumentOutOfRangeException()`, `EndRead_NullAsyncResult_ThrowsArgumentNullException()`, .....

```
CLR exception type: System.IO.IOException
    "Unable to read data from the transport connection: A request to send or receive data was disallowed because the socket had already been shut down in that direction with a previous shutdown call."
ntdll!ZwWaitForMultipleObjects+0x14:
00007ffa`d6e1f8d4 c3              ret
0:019> !pe
Exception object: 0000018fd824b8d0
Exception type:   System.IO.IOException
Message:          Unable to read data from the transport connection: A request to send or receive data was disallowed because the socket had already been shut down in that direction with a previous shutdown call.
InnerException:   System.Net.Sockets.SocketException, Use !PrintException 0000018fd824aee8 to see more.
StackTrace (generated):
    SP               IP               Function
    00000049660FC2E0 00007FFA3F2EBD07 System_Net_Sockets!System.Net.Sockets.NetworkStream.BeginRead(Byte[], Int32, Int32, System.AsyncCallback, System.Object)+0x447
    00000049660FE780 00007FFA3F2F1E24 System_Net_HttpListener!System.Net.HttpConnection.OnReadInternal(System.IAsyncResult)+0x394
    00000049660FE880 00007FFA3F2F19CE System_Net_HttpListener!System.Net.HttpConnection.OnRead(System.IAsyncResult)+0x9e
    00000049660FE8E0 00007FFA3F2F1532 System_Net_Sockets!System.Net.LazyAsyncResult.Complete(IntPtr)+0x292
    00000049660FE9E0 00007FFA3F2F1231 System_Net_Sockets!System.Net.ContextAwareResult.CompleteCallback()+0x71
    00000049660FEA20 00007FFA3F2F119F System_Net_Sockets!System.Net.ContextAwareResult+<>c.<Complete>b__17_0(System.Object)+0x6f
    00000049660FEA60 00007FFA70A837B1 System_Private_CoreLib!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)+0x111
    00000049660FEAD0 00007FFA3F2F0E57 System_Net_Sockets!System.Net.ContextAwareResult.Complete(IntPtr)+0x2e7
    00000049660FEBB0 00007FFA3F2F0B03 System_Net_Sockets!System.Net.LazyAsyncResult.ProtectedInvokeCallback(System.Object, IntPtr)+0x253
    00000049660FEC80 00007FFA3F2F0841 System_Net_Sockets!System.Net.LazyAsyncResult.InvokeCallback(System.Object)+0x31
    00000049660FECB0 00007FFA3F2EF5A3 System_Net_Sockets!System.Net.Sockets.BaseOverlappedAsyncResult.CompletionCallback(Int32, System.Net.Sockets.SocketError)+0x73
    00000049660FECF0 00007FFA3F2EF22E System_Net_Sockets!System.Net.Sockets.BaseOverlappedAsyncResult.CompletionPortCallback(UInt32, UInt32, System.Threading.NativeOverlapped*)+0x5ae
    00000049660FEEE0 00007FFA70C6DA43 System_Private_CoreLib!System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32, UInt32, System.Threading.NativeOverlapped*)+0x73

StackTraceString: <none>
HResult: 80131620
There are nested exceptions on this thread. Run with -nested for details
```

@davidsh @DavidGoll @CIPop

**Comment by caesar1995** [comment]
I'm working on this issue right now. This issue is critical for making tests working on UAP mode.

The approach I used to debug the test is:

- Focusing on one test file, remove other test files from .csproj
- After investigating that test file and make sure it runs clean, remove it from .csproj (to save time) and recover other test files 
- Investigating next test file. Repeat the process.

However, after I went through each test file and checkout the original test.csproj file (which includes all test files), this issue (`System.IO.IOException`) happened and broke the test runner again.

I did some experiments and found out that if some test cases are executing at the same time, the test runner would be broken. Similar to #21945 scenario. But I don't know which test cases need to run separately. 

This issue is blocking UAP work for S.N.HttpListener.

@CIPop @davidsh
